### ISSUE TITLE
jacocoTestCoverageVerification: The task was not up-to-date because it did not have any declared outputs.

### ISSUE THREAD

**Issue by C-Otto** [gradle/gradle#16589]
My project has several subprojects, and I run `jacocoTestCoverageVerification` for each subproject. According to the build scan, this task does cannot be cached, because it does not declare any outputs. Because of this I have to re-run the task for each subproject, which takes a significant amount of time.

Please declare outputs and/or enable build caching for `jacocoTestCoverageVerification`.

```
> Task :application:jacocoTestCoverageVerification
Caching disabled for task ':application:jacocoTestCoverageVerification' because:
  Caching has not been enabled for the task
Task ':application:jacocoTestCoverageVerification' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Excluding []
[ant:jacocoReport] Loading execution data file /somewhere/application/build/jacoco/integrationTest.exec
[ant:jacocoReport] Loading execution data file /somewhere/application/build/jacoco/test.exec
[ant:jacocoReport] Writing bundle 'application' with 3 classes
```

### Expected Behavior
Task is only run when necessary, i.e. if the previous run failed (due to lack of code coverage), or if any input changed.

### Current Behavior
It is always run.

### Your Environment
Gradle 6.8.3
Jacoco 0.8.6

**Comment by C-Otto** [comment]
Please unstale this issue. It's a bug, no matter how old it is.

**Comment by tkgregory** [comment]
Hi @C-Otto . I'm seeing a different behaviour. The `jacocoTestCoverageVerification`  task is marked as `UP-TO-DATE` when running with Gradle 7.4.2 and nothing has changed.

```
10:48:03: Executing 'jacocoTestCoverageVerification'...

> Task :compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava UP-TO-DATE
> Task :processTestResources NO-SOURCE
> Task :testClasses UP-TO-DATE
> Task :test UP-TO-DATE
> Task :jacocoTestReport UP-TO-DATE
> Task :jacocoTestCoverageVerification UP-TO-DATE

BUILD SUCCESSFUL in 98ms
5 actionable tasks: 5 up-to-date
10:48:04: Execution finished 'jacocoTestCoverageVerification'.
```

**Comment by C-Otto** [comment]
Could you try 6.8.3? That's the version I used when I opened this issue.

**Comment by C-Otto** [comment]
I can reproduce this with 6.8.3, 6.9, 6.9.1, 6.9.2, 7.0, 7.0.1, 7.0.2. It seems to be fixed with 7.1+. The release notes point me to #16939, which is a duplicate of this issue. At least it's fixed :)
